bl31/bl31.ld.S

```diff
@@ -224,9 +224,11 @@ SECTIONS
          */
         . = ALIGN(CACHE_WRITEBACK_GRANULE);
         __BAKERY_LOCK_START__ = .;
+        __PERCPU_BAKERY_LOCK_START__ = .;
         *(bakery_lock)
         . = ALIGN(CACHE_WRITEBACK_GRANULE);
-        __PERCPU_BAKERY_LOCK_SIZE__ = ABSOLUTE(. - __BAKERY_LOCK_START__);
+        __PERCPU_BAKERY_LOCK_END__ = .;
+        __PERCPU_BAKERY_LOCK_SIZE__ = __PERCPU_BAKERY_LOCK_END__ - __PERCPU_BAKERY_LOCK_START__;
```

Comment:
The linker symbols can be assigned relative values. The value of  `__PERCPU_BAKERY_LOCK_SIZE__ ` will not be correct after this change. 
According to LD documentation "a symbol created within a section definition is relative to the base of the section". Hence you need the `ABSOLUTE` function.

Reply:
No, I think the value computed will be wrong. If the difference between __PERCPU_BAKERY_LOCK_END__  and __PERCPU_BAKERY_LOCK_START__  is 0x10 , and the base of the section is 0x400000, then the value of `__PERCPU_BAKERY_LOCK_SIZE__ ` would be 0x400010.  The calculation following it would be wrong as well.